Rewrite lab 8 (#401)
puzzle/prometheus-training: content/en/docs/08/labs/81.md

title: "8.1 Tasks: Application Monitoring"
weight: 8
sectionnumber: 8.1
onlyWhenNot: baloise
---

### Task {{% param sectionnumber %}}.1: Create a ServiceMonitor

**Task description**:

Create a ServiceMonitor for the example application

* Create a ServiceMonitor, which will configure Prometheus to scrape metrics from the example-web-python application every 30 seconds.

>   * hint: `kubectl -n application-metrics apply -f my_file.yaml` will create a resource in the Kubernetes namespace

Reviewer: ```suggestion
  * hint: `{{% param cliToolName %}} -n application-metrics apply -f my_file.yaml` will create a resource in the Kubernetes namespace
```

* The example-web-python Service is labeled correctly and matches the labels you've defined in your ServiceMonitor.
* The port name in your ServiceMonitor configuration matches the port name in the Service definition.

>   * hint: check with `kubectl -n application-metrics get service example-web-python -o yaml`

Reviewer: ```suggestion
  * hint: check with `{{% param cliToolName %}} -n application-metrics get service example-web-python -o yaml`
```

> kubectl -n application-metrics apply -f ~/work/servicemonitor.yaml

Reviewer: ```suggestion
{{% param cliToolName %}} -n application-metrics apply -f ~/work/servicemonitor.yaml
```

```

> Verify that the target gets scraped in the [Prometheus user interface](http://LOCALHOST:19090/targets). Target name: `serviceMonitor/application-metrics/example-web-python-monitor/0` (It may take up to a minute for Prometheus to load the new configuration and scrape the metrics).

Reviewer: ```suggestion
Verify that the target gets scraped in the [Prometheus user interface](http://LOCALHOST:19090/targets). Target name: `serviceMonitor/application-metrics/example-web-python-monitor/0` (it may take up to a minute for Prometheus to load the new configuration and scrape the metrics).
```
